3party/opening_hours/osm_time_range.hpp

```diff
@@ -27,108 +27,491 @@
 #include <string>
 #include <vector>
 #include <iostream>
+#include <chrono>
+#include <type_traits>
 
 namespace osmoh
 {
-  class Time
+
+class Time
+{
+  enum EState
   {
-  public:
-    enum EFlags
-    {
-      eNone = 0,
-      eHours = 1,
-      eMinutes = 2,
-      ePlus = 4,
-      eMinus = 8,
-      eExt = 16,
-      eSunrise = 32,
-      eSunset = 64
-    };
-
-    uint8_t hours;
-    uint8_t minutes;
-    uint8_t flags;
-
-    Time() : hours(0), minutes(0), flags(eNone) {}
-    inline Time & Hours(uint8_t h) { hours = h; flags |= eHours; return *this; }
-    inline Time & Minutes(uint8_t m) { minutes = m; flags |= eMinutes; return *this; }
-    inline Time & Sunset() { flags = eSunset; return *this; }
-    inline Time & Sunrise() { flags = eSunrise; return *this; }
-
-    friend std::ostream & operator << (std::ostream & s, Time const & t);
+    eIsNotTime = 0,
+    eHaveHours = 1,
+    eHaveMinutes = 2,
   };
 
-  class TimeSpan
+  using TEStateRep = std::underlying_type<EState>::type;
+
+ public:
+  enum class EEvent
   {
-  public:
-    Time from;
-    Time to;
-    uint8_t flags;
-    Time period;
+    eNotEvent,
+    eDawn,
+    eSunrise,
+    eSunset,
+    eDusk
+  };
 
-    TimeSpan() : flags(Time::eNone) {}
+  using THours = std::chrono::hours;
+  using TMinutes = std::chrono::minutes;
 
-    friend std::ostream & operator << (std::ostream & s, TimeSpan const & span);
-  };
+ public:
+  Time() = default;
+  Time(Time const &) = default;
+  Time(THours const hours);
+  Time(TMinutes const minutes);
+  Time(THours const hours, TMinutes const minutes);
+  Time(EEvent const event);
 
-  class Weekdays
-  {
-  public:
-    uint8_t weekdays;
-    uint16_t nth;
-    int32_t offset;
+  Time & operator=(Time const &) = default;
 
-    Weekdays() : weekdays(0), nth(0), offset(0) {}
+  THours::rep GetHoursCount() const;
+  TMinutes::rep GetMinutesCount() const;
 
-    friend std::ostream & operator << (std::ostream & s, Weekdays const & w);
-  };
+  THours GetHours() const;
+  TMinutes GetMinutes() const;
+
+  void SetHours(THours const hours);
+  void SetMinutes(TMinutes const minutes);
+
+  EEvent GetEvent() const {return m_event;}
+  void SetEvent(EEvent const event);
+
+  bool IsEvent() const;
+  bool IsEventOffset() const;
+  bool IsHoursMinutes() const;
```

Comment:
I don't see definition of these methods, only declarations. Did you forget to add .cpp file to git index?